### PR TITLE
`Beam3D` can now solve for reaction loads

### PR DESCRIPTION
**Pull request by jashan498**
#### Brief description of what is fixed or changed
This PR includes the following changes:

- Changed `beam_3d` -> `Beam3D`.

- `solve_for_reaction_loads`now  solves for reaction loads instead of raising `NonImplementedError`.

- Included separate docstring for `Beam3D`'s `boundary_conditions` property.

- Added more test cases.

#### Other comments
ping @parsoyaarihant @moorepants 

#### Release Notes

* physics.continuum_mechanics
  * `Beam3D` can now solve for reaction loads

### CODE REVIEW

File: sympy/physics/continuum_mechanics/beam.py [hunk 1]
```diff
@@ -1252,7 +1253,38 @@ def moment_load_vector(self):
         """
         return self._moment_load_vector
 
-    def apply_load(self, value, dir="y", order=0):
+    @property
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        The dictionary has two kewwords namely slope and deflection.
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: keywords

File: sympy/physics/continuum_mechanics/beam.py [hunk 2]
```diff
@@ -1252,7 +1253,38 @@ def moment_load_vector(self):
...
+        Examples
+        ========
+        There is a beam of length 4 meters. The slope at 0 should be 4 along
+        the x-axis and 0 along others. At the other end of beam, deflection
+        along all the three axes should be zero.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam3D
+        >>> from sympy import symbols
+        >>> l, E, G, I, A, x = symbols('l, E, G, I, A, x')
+        >>> b = Beam3D(30, E, G, I, A, x)
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Doesn't a 3d beam have to have two second moments of area and a product moment of area?

Comment (moorepants) [reply sympy/physics/continuum_mechanics/beam.py]: What about the second polar moment of area? Is this calculated from I somehow?

**Review by moorepants — CHANGES_REQUESTED** [review]
One comment needs to be addressed.

I'm not super confident this class is complete and correct. We need more comprehensive examples/test cases to show it works.

File: sympy/physics/continuum_mechanics/beam.py (hunk 2 above)
Comment (jashan498) [reply sympy/physics/continuum_mechanics/beam.py]: Yeah, 3d beam has to have two second moments. It's just if `Ix = Iy`, we can pass only one value (Like `I` here) and both moments would get equated to it. And if they are unequal, then pass a `list` of the form `[Ix, Iy]` (Mentiond [here](https://github.com/sympy/sympy/blob/ff402664c8753c886c97335aa84635d1de3fe97e/sympy/physics/continuum_mechanics/beam.py#L1522) in the docstring).

Comment (jashan498) [reply sympy/physics/continuum_mechanics/beam.py]: There is no special method for polar moment as of now. I think it would be just equal to `Ix + Iy`?

**Review by moorepants — APPROVED** [review]

**Comment by moorepants** [comment]
I'm going to merge this, but can you open an issue requesting that more examples and tests be made for this class.

**Pull request merged**